tests/attr/test_guided_backprop.py

```diff
@@ -120,7 +121,8 @@ def _guided_backprop_matching_assert(
         test_input: TensorOrTupleOfTensorsGeneric,
     ):
         out = model(test_input)
-        attrib = GuidedBackprop(model)
+        model_copy = copy.deepcopy(model)
```

Comment:
nit: why do we do `deepcopy` here ?

Reply:
deepcopy is used here since once full_backward_hooks are placed on a module, then standard backward hooks can no longer be added. Since this is using neuron guided backprop (with layer output), this falls back to standard backward hooks within NeuronGuidedBackprop but only full backward hooks in GuidedBackprop, requiring a copy of the model rather than using the same model.